typing/ctype.ml

```diff
@@ -1838,6 +1838,28 @@ let enter_poly env univar_pairs t1 tl1 t2 tl2 f =
 
 let univar_pairs = ref []
 
+let reify_univars env ty =
+  let rec subst_univar vars ty =
+    let ty = repr ty in
+    if ty.level >= lowest_level then begin
+      ty.level <- pivot_level - ty.level;
+      match ty.desc with
+      | Tvar name ->
+          save_desc ty ty.desc;
+          let t = newty2 ty.level (Tunivar name) in
+          vars := t :: !vars;
+          ty.desc <- Tsubst t
+      | _ ->
+          iter_type_expr (subst_univar vars) ty
+    end
+  in
+  let vars = ref [] in
+  subst_univar vars ty;
+  unmark_type ty;
+  let ty = copy ~env ty in
+  cleanup_types ();
+  newty2 ty.level (Tpoly(repr ty, !vars))
```

Comment:
It took me a while to realize that the `ty.level` here is about the fresh copy...
I would suggest adding a comment at the beginning of the function that the argument to `reify_univars` is supposed to be at `generic_level` (i.e. it comes from a generalized type declaration). Otherwise this code would not work. Under this assumption the last `newty2` could be just `newty`, at the level on `ty` will be `current_level` anyway (not essential).